src/App.tsx

```diff
@@ -1,22 +1,57 @@
-import React from 'react';
-import { useSelector } from 'react-redux';
+import React, { useMemo } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
 
 import './App.scss';
-import { Start } from './components/Start';
 
-import { isLoading, getMessage } from './store';
+import {
+  getLoading, getPosts, getError, getInitialized,
+} from './redux/selectors';
+import { loadPosts } from './redux/store';
+import { PostsList } from './components/PostsList';
 
 const App = () => {
-  const loading = useSelector(isLoading);
-  const message = useSelector(getMessage) || 'Ready!';
+  const isLoading = useSelector(getLoading);
+  const isInitialized = useSelector(getInitialized);
+  const posts = useSelector(getPosts);
+  const hasError = useSelector(getError);
+  const dispatch = useDispatch();
 
-  return (
-    <div className="App">
-      <h1>Redux list of posts</h1>
-      <h2>{loading ? 'Loading...' : message}</h2>
+  const loadData = () => {
+    dispatch(loadPosts());
+  };
+
+  const isButtonVisible = useMemo(() => !isLoading && !isInitialized,
+    [isLoading, isInitialized]);
+  const shouldPostsRender = useMemo(() => !hasError && !isLoading && isInitialized,
+    [hasError, isLoading, isInitialized]);
```

Comment:
refactor code here as I showed you above

Reply:
✅